src/Adapter/Address/AddressFormatter.php

```diff
@@ -0,0 +1,45 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Address;
+
+use Address;
+use AddressFormat;
+use PrestaShop\PrestaShop\Core\Address\AddressFormatterInterface;
+
+class AddressFormatter implements AddressFormatterInterface
+{
+    public function generateAddress(int $addressId): array
+    {
+        $formatted = AddressFormat::generateAddress(
```

Comment:
If you check inside AddressFormat::generateAddress, the code uses string concatenation to build the string, inserting separator between each item.

Then you explode the generated string, reversing the process.

In order to avoid this "I do X then I revert X", maybe we could improve AddressFormat::generateAddress ?

```
public static function generateAddress(Address $address, $patternRules = [], $newLine = self::FORMAT_NEW_LINE, $separator = ' ', $style = [])
{
    $eachParts = self::generateAddressParts($address, $patternRules, $newLine, $separator, $style);
    return $this->concatenateParts($eachParts);
}
```

Then you could use `AddressFormat::generateAddressParts()` (the name must be improved) and avoid relying on `explode()`.